src/main/java/thePackmaster/actions/siegepack/ShrapnelAction.java

```diff
@@ -0,0 +1,54 @@
+package thePackmaster.actions.siegepack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.powers.VulnerablePower;
+import thePackmaster.powers.shamanpack.IgnitePower;
+
+import static com.megacrit.cardcrawl.dungeons.AbstractDungeon.player;
+
+//REFS : LifeDrainAction (bardinspirepack), Venemous stance (serpentinepack).
+public class ShrapnelAction extends AbstractGameAction {
+    private final int DAMAGE_THRESHOLD;
+    private final int DAMAGE;
+    private final int POWER_STACKS;
+
+    public ShrapnelAction(AbstractCreature target, AbstractCreature source, int amount, int threshold, DamageInfo.DamageType type, int stacks)
+    {
+        setValues(target, source, amount);  //Shorthand. Also sets duration = 0.5F
+        duration = 0.1F;
+        DAMAGE = amount;
+        actionType = AbstractGameAction.ActionType.DAMAGE;
+        damageType = type;
+        POWER_STACKS = stacks;
+        DAMAGE_THRESHOLD = threshold;
+    }
+
+    @Override
+    public void update()
+    {
+        //In any Action, must set "isDone" to "true" or call tickDuration(), else Crash.
+        isDone = true;
+        if (damageType != DamageInfo.DamageType.NORMAL || DAMAGE < DAMAGE_THRESHOLD)
+        { return; }
+
+        applyDebuffs(DAMAGE);
+        if (AbstractDungeon.getMonsters().areMonstersBasicallyDead()) {
+            AbstractDungeon.actionManager.clearPostCombatActions();
+        }
+        AbstractDungeon.actionManager.addToTop(new WaitAction(0.1f));
+    }
+
+    private void applyDebuffs (int damageAmount) {
+        //Apply debuffs for each 10 unblocked damage dealt by the attack.
+        int triggerCount = damageAmount / DAMAGE_THRESHOLD;
+        AbstractDungeon.actionManager.addToTop(new ApplyPowerAction(target, player, new IgnitePower(target, 2 * POWER_STACKS * triggerCount), 2 * POWER_STACKS * triggerCount, true));
```

Comment:
Might want to check if trigger count is > 0 in case that logic isn't somewhere else already

Reply:
![threshold check](https://github.com/user-attachments/assets/8a7c8d5c-d406-4ae7-a062-b94f0bf4c85b)
It's handled at the start of the update(), this is good, no? In-game behavior in tests is as intended but maybe there's a better way to code this

Reply:
Oh yeah, makes sense, all good then